certbot/certbot/_internal/client.py

```diff
@@ -528,6 +534,22 @@ def obtain_and_enroll_certificate(self, domains: List[str], certname: Optional[s
             key.pem, chain,
             self.config)
 
+    def _retry_obtain_certificate(self, error: messages.Error, key: util.Key,
+                                csr: util.CSR, domains: List[str],
+                                ) -> Tuple[bytes, bytes, util.Key, util.CSR]:
+        if self.config.allow_subset_of_names and error.subproblems is not None:
+            failed_domains = [problem.identifier.value for problem in error.subproblems
+                                if problem.identifier is not None]
+            successful_domains = [x for x in domains if x not in failed_domains]
+            if successful_domains != domains and len(successful_domains) != 0:
+                display_util.notify("Unable to obtain a certificate with every requested "
+                    f"domain. Retrying without: {failed_domains}")
```

Comment:
It would be ideal if `failed_domains` could be printed in a prettier way than the default `str` representation of Python lists.

`display_util.summarize_domain_list` is one option, but I'm open to others, if we don't want to abbreviate the domains that are being skipped.

Reply:
I'm happy to use `summarize_domain_list` if you're ok with that tradeoff when more than 2 domains fail.